File: QE selenium java/Activity8.java
package demos;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.interactions.Actions;

public class Activity8 {
    public static void main(String[] args) {

        WebDriver driver = new FirefoxDriver();
        Actions actions = new Actions(driver);

        driver.get("https://training-support.net/webelements/mouse-events");

        System.out.println("Page Title: " + driver.getTitle());

        WebElement cargoLock = driver.findElement(By.id("cargo-lock"));
        WebElement cargoToml = driver.findElement(By.id("cargo-toml"));
        WebElement src = driver.findElement(By.id("src"));
        WebElement target = driver.findElement(By.id("target"));
        WebElement confirmation = driver.findElement(By.id("result"));

        actions
                .click(cargoLock)
                .moveToElement(cargoToml)
                .click()
                .build()
                .perform();

        System.out.println("Confirmation Text (Seq 1): " + confirmation.getText());

        actions
                .doubleClick(src)
                .contextClick(target)
                .build()
                .perform();

        driver.findElement(By.cssSelector("li[data-action='open']")).click();
        System.out.println("Confirmation Text (Seq 2): " + confirmation.getText());
        driver.quit();
    }
}

File: QE-selenium python/Activity8.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.common.action_chains import ActionChains


driver = webdriver.Firefox()
actions = ActionChains(driver)

driver.get("https://training-support.net/webelements/mouse-events")

print("Page Title:", driver.title)

cargo_lock = driver.find_element(By.ID, "cargo-lock")
cargo_toml = driver.find_element(By.ID, "cargo-toml")
src = driver.find_element(By.ID, "src")
target = driver.find_element(By.ID, "target")
confirmation = driver.find_element(By.ID, "result")

actions.click(cargo_lock)\
       .move_to_element(cargo_toml)\
       .click()\
       .perform()

print("Confirmation Text (Seq 1):", confirmation.text)

actions.double_click(src)\
       .context_click(target)\
       .perform()

driver.find_element(By.CSS_SELECTOR, "li[data-action='open']").click()

print("Confirmation Text (Seq 2):", confirmation.text)

driver.quit()
